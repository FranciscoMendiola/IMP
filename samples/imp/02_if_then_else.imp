if x <= 10 then
  y := 100
else
  y := 200
