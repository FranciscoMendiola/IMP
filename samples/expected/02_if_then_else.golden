Token {tokenKind = KIf, lexeme = "if", position = 0}
Token {tokenKind = Id, lexeme = "x", position = 3}
Token {tokenKind = Leq, lexeme = "<=", position = 5}
Token {tokenKind = IntLit, lexeme = "10", position = 8}
Token {tokenKind = KThen, lexeme = "then", position = 11}
Token {tokenKind = Id, lexeme = "y", position = 18}
Token {tokenKind = Assign, lexeme = ":=", position = 20}
Token {tokenKind = IntLit, lexeme = "100", position = 23}
Token {tokenKind = KElse, lexeme = "else", position = 27}
Token {tokenKind = Id, lexeme = "y", position = 34}
Token {tokenKind = Assign, lexeme = ":=", position = 36}
Token {tokenKind = IntLit, lexeme = "200", position = 39}
